cranelift/filetests/src/function_runner.rs

```diff
@@ -297,13 +297,16 @@ fn make_trampoline(signature: &ir::Signature, isa: &dyn TargetIsa) -> Function {
         .enumerate()
         .map(|(i, param)| {
             // Calculate the type to load from memory, using integers for booleans (no encodings).
-            let ty = if param.value_type.is_bool() {
-                Type::int(max(param.value_type.bits(), 8)).expect(
-                    "to be able to convert any boolean type to its equal-width integer type",
-                )
+            let is_scalar_bool = param.value_type.is_bool();
+            let is_vector_bool =
+                param.value_type.is_vector() && param.value_type.lane_type().is_bool();
+
+            let ty = if is_scalar_bool || is_vector_bool {
+                param.value_type.as_int()
```

Comment:
Can you turn this sequence into a helper on the type itself? Something like `param.value_type.coerce_bools_to_ints()` would be reasonable, IMHO...